pandas/core/generic.py

```diff
@@ -2801,14 +2801,16 @@ def notnull(self):
         """
         return notnull(self).__finalize__(self)
 
-    def clip(self, lower=None, upper=None, out=None):
+    def clip(self, lower=None, upper=None, out=None, axis=None):
         """
         Trim values at input threshold(s)
 
         Parameters
         ----------
-        lower : float, default None
-        upper : float, default None
+        lower : float or array_like, default None
+        upper : float or array_like, default None
+        axis : int, optional
```

Comment:
Grepping through the files, it looks like most docstrings mention only integers, but I agree that your suggestion is better.